buildtools/npm/manifest.go

```diff
@@ -35,3 +36,34 @@ type Lockfile struct {
 func FromLockfile(filename string) (Lockfile, error) {
 	return Lockfile{}, errors.ErrNotImplemented
 }
+
+func convertManifestToPkg(manifest manifest) pkg.Package {
+	id := pkg.ID{
+		Type:     pkg.NodeJS,
+		Name:     manifest.Name,
+		Revision: manifest.Version,
+	}
+
+	var imports pkg.Imports
+	for depName, version := range manifest.Dependencies {
+		imports = append(imports, createImport(depName, version))
+	}
+
+	return pkg.Package{
+		ID:      id,
+		Imports: imports,
+	}
+}
+
+func createImport(packageName string, version string) pkg.Import {
```

Comment:
I would inline this unless there's a good reason to separate it e.g.:

- This is called in multiple places.
- This is an independently tested function.

Descriptive function names are valuable, but I think comments work better in cases where there's no control flow reason to extract a function.